test(integration): actually test a few things in Electron

eventualbuddha opened the pull request:
There are tools that are better suited to running tests in Electron, but they're all kinda heavy (see https://webdriver.io). So instead, I abuse `vite` for fun and profit! Uses a custom vite plugin to create a test server for both development and CI. It's actually kinda great: we can use TypeScript and automatically rebuild/retest on file save in development, and in CI it runs once and exits:

```
$ ./test/integration/run
[129](https://github.com/votingworks/kiosk-browser/runs/6516761961?check_suite_focus=true#step:9:130)
[browser] [3387:0519/235458.409214:ERROR:bus.cc(397)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[130](https://github.com/votingworks/kiosk-browser/runs/6516761961?check_suite_focus=true#step:9:131)
[browser] [3387:0519/235458.409272:ERROR:bus.cc(397)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[131](https://github.com/votingworks/kiosk-browser/runs/6516761961?check_suite_focus=true#step:9:132)
[browser] [3387:0519/235458.481255:ERROR:bus.cc(397)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[132](https://github.com/votingworks/kiosk-browser/runs/6516761961?check_suite_focus=true#step:9:133)
[browser] [3426:0519/235458.525602:ERROR:viz_main_impl.cc(188)] Exiting GPU process due to errors during initialization
[133](https://github.com/votingworks/kiosk-browser/runs/6516761961?check_suite_focus=true#step:9:134)
[browser] [3457:0519/235458.602617:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
[134](https://github.com/votingworks/kiosk-browser/runs/6516761961?check_suite_focus=true#step:9:135)
[server] ✅ storage
[135](https://github.com/votingworks/kiosk-browser/runs/6516761961?check_suite_focus=true#step:9:136)
[server] ✅ file system
[136](https://github.com/votingworks/kiosk-browser/runs/6516761961?check_suite_focus=true#step:9:137)
[server] ✅ battery
[137](https://github.com/votingworks/kiosk-browser/runs/6516761961?check_suite_focus=true#step:9:138)
[server] ✅ print to PDF
[138](https://github.com/votingworks/kiosk-browser/runs/6516761961?check_suite_focus=true#step:9:139)
[server] 
[139](https://github.com/votingworks/kiosk-browser/runs/6516761961?check_suite_focus=true#step:9:140)
[server] 4 passed
[140](https://github.com/votingworks/kiosk-browser/runs/6516761961?check_suite_focus=true#step:9:141)
```

🙌 🙌 🙌 

Closes #118

jonahkagan reviewed (commented):
I didn't quite grok what's going on here - I would probably need to look up all the commands in test/integration/run to figure it out. I'm happy to do that, though some explanatory comments might make it easier, but I don't have time before I leave for vacation. So someone else might need to review this.

But I wanted to leave one piece of feedback, which is that I'm wondering if there's any way to construct these tests using any of our existing testing tools (e.g. jest/cypress)? While this is really cool and small now, I can imagine as we build on it we'll end up wanting more features, and there having a familiar tool might help save us time. Not a blocker, just a provocation!

eventualbuddha commented:
Cypress experimented with Electron app support [but discontinued it (see note)](https://www.cypress.io/blog/2019/09/26/testing-electron-js-applications-using-cypress-alpha-release/). While they do test their own app with Cypress, they do it by stubbing out a bunch of the Electron stuff. I don't think that helps us.

Jest could be used with something like [jest-puppeteer](https://github.com/smooth-code/jest-puppeteer). The downside to that approach is that the Jest tests are running in Node and not in the browser, similar to Cypress. That makes it easier to test an application that's using `kiosk-browser`, but doesn't let us put assertions in the running page environment. We could write a sample application/page that exercises the features we want to test, and then write tests that interact with that application and makes sure the webpage contents change as expected. This seems unnecessarily indirect, though.

I share the concern that this could grow in size and complexity over time in such a way that it'll cause problems for us. To that end, I switched `minitest` to use Jest's `expect` and to add the `before`/`after` hooks. I also added a decent amount of comments & docs. Hopefully that eases your concerns!

jonahkagan commented:
No objection here! Agreed with @benadida's take
